dialog-polyfill.js

```diff
@@ -115,6 +121,26 @@
     },
 
     /**
+     * Maybe assign a default role attribute to this dialog that is being
+     * polyfilled. Without one, the dialog won't have a useful a11y tree generated
+     * for its core usage in a Web application.
+     */
+    maybeSetADefaultRoleValue: function(){
+      if ( !this.dialog.hasAttribute('role') || this.dialog.getAttribute('role') === ''  ) {
+        this.dialog.setAttribute( 'role', 'dialog'  );
```

Comment:
Hmm with the test runs I had, it only sets `role="dialog" if no role
attribute has been set or if it's empty; if it's set with a value by an end
user (developer), it shouldn't disregard what unique role they've decided
to explicitly use.

Did your test runs of the code  being executed show a deviation from this?

On Fri, Feb 5, 2016 at 5:44 PM, Sam Thorogood <notifications@github.com>
wrote:

> In dialog-polyfill.js
> <https://github.com/GoogleChrome/dialog-polyfill/pull/92#discussion_r52083109>
> :
>
> > @@ -115,6 +121,26 @@
> >      },
> >
> >      /**
> > +     * Maybe assign a default role attribute to this dialog that is being
> > +     * polyfilled. Without one, the dialog won't have a useful a11y tree generated
> > +     * for its core usage in a Web application.
> > +     */
> > +    maybeSetADefaultRoleValue: function(){
> > +      if ( !this.dialog.hasAttribute('role') || this.dialog.getAttribute('role') === ''  ) {
> > +        this.dialog.setAttribute( 'role', 'dialog'  );
>
> So this approach will just eventually set the role="dialog" on each
> dialog. It will never transition back to not being a dialog.
>
> With that in mind, I'd like to suggest #95
> <https://github.com/GoogleChrome/dialog-polyfill/pull/95> as just a quick
> fix since it's basically paramount to this approach :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleChrome/dialog-polyfill/pull/92/files#r52083109>.
>



-- 
Kevin Lozandier
lozandier@gmail.com <lozandier@gmail.com>